kivy/utils.py

```diff
@@ -110,17 +110,34 @@ def rgba(s, *args):
     raise Exception('Invalid value (not a string / list / tuple)')
 
 
-def get_color_from_hex(s):
+def get_color_from_hex(color):
     '''Transform a hex string color to a kivy
     :class:`~kivy.graphics.Color`.
+
+    .. versionchanged:: 2.2.0
+
+    Supporting and 3-4 HEX values.
+
+    >>> get_color_from_hex("#FF00FF33")  # [1.0, 0.0, 1.0, 0.2]
+    >>> get_color_from_hex("#F0F3")      # [1.0, 0.0, 1.0, 0.2]
+    >>> get_color_from_hex("#F0F")       # [1.0, 0.0, 1.0, 1.0]
     '''
-    if s.startswith('#'):
-        return get_color_from_hex(s[1:])
+    # Remove the usually HEX prefix
+    if color.startswith("#"):
+        color = color[1:]
 
+    # In case the color have 3-4 values, double the values
+    if 2 < len(color) < 5:
+        color = f"{'{}' * len(color)}".format(*[value * 2 for value in color])
```

Comment:
Maybe we can also add a check which asserts if the string length does not look good?

Reply:
English is not my first language, so can you be more descriptive? Maybe with an example. 😕

Reply:
Did you mean to provide an descriptive error message in case the provided color have in invalid format or it have less values then 3-character or more values then 8-character?